packages/gateway/src/features/application/root-resolvers.ts

```diff
@@ -0,0 +1,99 @@
+import { hasScope } from '@gateway/features/user/utils'
+import { IApplicationConfigPayload } from '@gateway/features/application/type-resolvers'
+
+export const resolvers: GQLResolver = {
+  Mutation: {
+    async updateApplicationConfig(_, { applicationConfig = {} }, authHeader) {
+      // Only natlsysadmin should be able to update application config
+      if (!hasScope(authHeader, 'natlsysadmin')) {
+        return await Promise.reject(
+          new Error(
+            'Update application config is only allowed for natlsysadmin'
+          )
+        )
+      }
+      const applicationConfigPayload: IApplicationConfigPayload = {
+        APPLICATION_NAME: applicationConfig.APPLICATION_NAME as string,
+        BACKGROUND_SYNC_BROADCAST_CHANNEL:
+          applicationConfig.BACKGROUND_SYNC_BROADCAST_CHANNEL as string,
+        COUNTRY: applicationConfig.COUNTRY as string,
+        COUNTRY_LOGO_FILE: applicationConfig.COUNTRY_LOGO_FILE as string,
+        COUNTRY_LOGO_RENDER_WIDTH:
+          applicationConfig.COUNTRY_LOGO_RENDER_WIDTH as number,
+        COUNTRY_LOGO_RENDER_HEIGHT:
+          applicationConfig.COUNTRY_LOGO_RENDER_HEIGHT as number,
+        DESKTOP_TIME_OUT_MILLISECONDS:
+          applicationConfig.DESKTOP_TIME_OUT_MILLISECONDS as number,
+        LANGUAGES: applicationConfig.LANGUAGES as string,
+        CERTIFICATE_PRINT_CHARGE_FREE_PERIOD:
+          applicationConfig.CERTIFICATE_PRINT_CHARGE_FREE_PERIOD as number,
+        CERTIFICATE_PRINT_CHARGE_UP_LIMIT:
+          applicationConfig.CERTIFICATE_PRINT_CHARGE_UP_LIMIT as number,
+        CERTIFICATE_PRINT_LOWEST_CHARGE:
+          applicationConfig.CERTIFICATE_PRINT_LOWEST_CHARGE as number,
+        CERTIFICATE_PRINT_HIGHEST_CHARGE:
+          applicationConfig.CERTIFICATE_PRINT_HIGHEST_CHARGE as number,
+        UI_POLLING_INTERVAL: applicationConfig.UI_POLLING_INTERVAL as number,
+        FIELD_AGENT_AUDIT_LOCATIONS:
+          applicationConfig.FIELD_AGENT_AUDIT_LOCATIONS as string,
+        APPLICATION_AUDIT_LOCATIONS:
+          applicationConfig.APPLICATION_AUDIT_LOCATIONS as string,
+        INFORMANT_MINIMUM_AGE:
+          applicationConfig.INFORMANT_MINIMUM_AGE as number,
+        HIDE_EVENT_REGISTER_INFORMATION:
+          applicationConfig.HIDE_EVENT_REGISTER_INFORMATION as boolean,
+        EXTERNAL_VALIDATION_WORKQUEUE:
+          applicationConfig.EXTERNAL_VALIDATION_WORKQUEUE as boolean,
+        SENTRY: applicationConfig.SENTRY as string,
+        LOGROCKET: applicationConfig.LOGROCKET as string,
+        PHONE_NUMBER_PATTERN:
+          applicationConfig.PHONE_NUMBER_PATTERN as GQLPhoneNumberPatternInput,
+        BIRTH_REGISTRATION_TARGET:
+          applicationConfig.BIRTH_REGISTRATION_TARGET as number,
+        DEATH_REGISTRATION_TARGET:
+          applicationConfig.DEATH_REGISTRATION_TARGET as number,
+        NID_NUMBER_PATTERN:
+          applicationConfig.NID_NUMBER_PATTERN as GQLNIDNumberPatternInput
+      }
```

Comment:
In this way, it only saves the value that is being passed to this mutation. So that we don't have to make multiple mutations. And it's working as intended.